aws-predictions/src/main/java/com/amplifyframework/predictions/aws/http/LivenessWebSocket.kt

```diff
@@ -392,44 +393,63 @@ internal class LivenessWebSocket(
             width = initialFaceRect.width() / clientSessionInformation.videoWidth
         )
         faceDetectedStart = adjustedDate(videoStartTime)
-        val clientInfoEvent =
-            ClientSessionInformationEvent(
+
+        if (challengeType == null) {
+            onErrorReceived.accept(
+                PredictionsException(
+                    "Failed to send an initial face detected event",
+                    AmplifyException.TODO_RECOVERY_SUGGESTION
+                )
+            )
+        } else {
+            val clientInfoEvent =
+                ClientSessionInformationEvent(
+                    challenge = buildClientChallenge(
+                        challengeType = challengeType!!,
+                        challengeId = challengeId,
+                        initialFace = InitialFace(
+                            boundingBox = initialDetectedFace!!,
+                            initialFaceDetectedTimestamp = faceDetectedStart
+                        ),
+                        videoStartTimestamp = videoStartTimestamp
+                    )
+                )
+            sendClientInfoEvent(clientInfoEvent)
+        }
+    }
+
+    fun sendFinalEvent(targetFaceRect: RectF, faceMatchedStart: Long, faceMatchedEnd: Long) {
```

Comment:
create a local challengeType variable here to remove !! later on in this method.